gen_code_can_common.c

```diff
@@ -0,0 +1,820 @@
+    output->data[4] = quality; 
+
+    
+    output->data_len = 5;
+
+    return true;
+}
+        
+bool build_fill_lvl_msg(uint32_t timestamp,
+                                                   uint8_t level, 
+       enum FILL_DIRECTION fill_direction, 
+
+                        can_msg_t *output)
+
+{
+    if (!output) { return false; }
+
+    output->sid = MSG_GPS_ALTITUDE | BOARD_UNIQUE_ID;
+    write_timestamp_3bytes(timestamp, output);        
+    
+        output->data[3] = level; 
+    output->data[4] = (uint8_t) fill_direction; 
+
+    
+    output->data_len = 5;
+
+    return true;
+}
+        
+bool build_radi_value_msg(uint32_t timestamp,
+                                                   uint8_t radi_board, 
+                           uint16_t radi, 
+
+                        can_msg_t *output)
+
+{
+    if (!output) { return false; }
+
+    output->sid = MSG_GPS_ALTITUDE | BOARD_UNIQUE_ID;
+    write_timestamp_3bytes(timestamp, output);        
+    
+        output->data[3] = radi_board; 
+    output->data[4] = radi; 
+
+    
+    output->data_len = 5;
+
+    return true;
+}
+        
+int get_general_cmd(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+int get_actuator_cmd(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+bool get_alt_arm_cmd(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    if (!altimeter) { return false; } 
+
+
+}
+
+int get_reset_cmd(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+bool get_debug_msg(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    if (!level) { return false; } 
+    if (!line) { return false; } 
+
+
+}
+
+bool get_debug_printf(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+bool get_debug_radio_cmd(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+int get_actuator_status(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+bool get_alt_arm_status(const can_msg_t *msg,
+                                                   uint16_t drogue_v, 
+                           uint16_t main_v, 
+)
+
+{
+    if (!msg){ return false; }
+    if (!altimeter) { return false; } 
+    if (!drogue_v) { return false; } 
+    if (!main_v) { return false; } 
+
+
+}
+
+int get_general_board_status(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+bool get_sensor_temp(const can_msg_t *msg,
+                        uint8_t sensor_id, 
+                           uint24_t temperature, 
+)
+
+{
+    if (!msg){ return false; }
+    if (!sensor_id) { return false; } 
+    if (!temperature) { return false; } 
+
+
+}
+
+bool get_sensor_altitude(const can_msg_t *msg,
```

Comment:
name: get_altitude_data
params: `uint32_t altitude` instead of `int32 *altitude`
missing:
` if (get_message_type(msg) != MSG_SENSOR_ALTITUDE) { return false; }

    *altitude = ((uint32_t)msg->data[3] << 24);
    *altitude |= ((uint32_t)msg->data[4] << 16);
    *altitude |= ((uint32_t)msg->data[5] << 8);
    *altitude |= msg->data[6];

    return true;`